src/polynomial.rs

```diff
@@ -59,6 +60,69 @@ impl Polynomial {
     pub fn to_vec(&self) -> Vec<Fr> {
         self.elements.clone()
     }
+    pub fn ifft_on_elements(&mut self) -> Result<(), PolynomialError> {
+        let ifft_result = Self::ifft(&self.to_vec());
+        match ifft_result {
+            Ok(ifft_result) => {
+                self.elements = ifft_result;
+                Ok(())
+            },
+            Err(e) => Err(e),
+        }
+    }
+
+    pub fn fft_on_elements(&mut self) -> Result<(), PolynomialError> {
+        let fft_result = Self::fft(&self.to_vec());
+        match fft_result {
+            Ok(fft_result) => {
+                self.elements = fft_result;
+                Ok(())
+            },
+            Err(e) => Err(e),
+        }
+    }
+
+    pub fn ifft(vals: &Vec<Fr>) -> Result<Vec<Fr>, PolynomialError> {
+        let length = vals.len();
+
+        if !length.is_power_of_two() {
+            return Err(PolynomialError::FFTError(format!(
+                "Length of fr vec provided is not a power of 2: {}",
+                vals.len()
+            )));
+        }
+
+        match GeneralEvaluationDomain::<Fr>::new(length) {
+            Some(domain) => {
+                let result = domain.ifft(vals);
+                Ok(result)
+            },
+            None => Err(PolynomialError::FFTError(
+                "Failed to construct domain for IFFT".to_string(),
+            )),
+        }
+    }
+
+    pub fn fft(vals: &Vec<Fr>) -> Result<Vec<Fr>, PolynomialError> {
+        let length = vals.len();
+
+        if !length.is_power_of_two() {
```

Comment:
This check isn't needed. Since all methods of creating an instance of the struct forces the elements to be a power of 2, this is enforced at the creation level.

Reply:
check removed as a part of df504078e06c54f48c23b1006916455fe75fd0a0